contracts/base/ERC721Base.sol

```diff
@@ -0,0 +1,144 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.0;
+
+import { ERC721A } from "../eip/ERC721A.sol";
+
+import "../extension/ContractMetadata.sol";
+import "../extension/Multicall.sol";
+import "../extension/Ownable.sol";
+import "../extension/RoyaltyUpdated.sol";
+import "../extension/BatchMintMetadata.sol";
+
+import "../lib/TWStrings.sol";
+
+/**
+ *  The `ERC721Base` smart contract implements the ERC721 NFT standard, along with the ERC721A optimization to the standard.
+ *  It includes the following additions to standard ERC721 logic:
+ *
+ *      - Ability to mint NFTs via the provided `mint` function.
+ *
+ *      - Contract metadata for royalty support on platforms such as OpenSea that use
+ *        off-chain information to distribute roaylties.
+ *
+ *      - Ownership of the contract, with the ability to restrict certain functions to
+ *        only be called by the contract's owner.
+ *
+ *      - Multicall capability for fetching NFT data.
```

Comment:
nit: multicall is actually needed for updating royalty+contract metadata in one go. Maybe say `Multicall capability to perform multiple actions atomically` or something like that